### PR TITLE
add pyspark hive support

### PR DESCRIPTION
**Pull request by hezhaozhao-git**
This is the hive support of spark on ray, the related issue is :[raydp not support read hive](https://github.com/oap-project/raydp/issues/220) ,I found that the original implementation does not support hive read and write by default, this is my contribution,If there is any problem, welcome to correct and discuss

### CODE REVIEW

**Comment by hezhaozhao-git** [comment]
It's not my code bug, what should I do @ConeyLiu @kira-lin

File: python/raydp/context.py [hunk 1]
```diff
@@ -56,13 +60,15 @@ def __init__(self,
                  num_executors: int,
                  executor_cores: int,
                  executor_memory: Union[str, int],
+                 enable_hive: Optional[bool],
```
Comment (kira-lin) [inline python/raydp/context.py:63]: No need to use `Optional` here, because it's never None, just False or True.

**Review by kira-lin — CHANGES_REQUESTED** [review]
Hi @hezhaozhao-git 
Thanks for submitting this PR, we appreciate your contribution!
CI failed due to lint error. Can you please fix this:
python/raydp/spark/ray_cluster.py:86:0: C0301: Line too long (112/100) (line-too-long)
and other review comments. Thanks

File: python/raydp/spark/ray_cluster.py [hunk 2]
```diff
@@ -80,9 +81,13 @@ def get_spark_session(self,
...
-        self._spark_session =\
-            spark_builder.appName(app_name).master(self.get_cluster_url()).getOrCreate()
-        return self._spark_session
+        if enable_hive:
+            self._spark_session = \
+                spark_builder.appName(app_name).master(self.get_cluster_url()).enableHiveSupport().getOrCreate()
+        else:
+            self._spark_session = \
+                spark_builder.appName(app_name).master(self.get_cluster_url()).getOrCreate()
+            return self._spark_session
```
[the remark this answers is not in the document]
Comment (hezhaozhao-git) [reply python/raydp/spark/ray_cluster.py:90]: OK, I'll fix these issues right away

File: python/raydp/context.py (hunk 1 above)
Comment (hezhaozhao-git) [reply python/raydp/context.py:63]: yes

File: python/raydp/context.py [hunk 3]
```diff
@@ -37,6 +37,10 @@ class _SparkContext(ContextDecorator):
     :param num_executors the number of executor requested
     :param executor_cores the CPU cores for each executor
     :param executor_memory the memory size (eg: 10KB, 10MB..) for each executor
+    :param enable_hive: spark read hive data source：If you want to use this function,
+                    please install the corresponding spark version first, set ENV SPARK_HOME,
+                    configure spark-env.sh HADOOP_CONF_DIR in spark conf, and copy hive-site.xml
+                    and hdfs-site.xml to ${SPARK_HOME}/ conf
```
[the remark this answers is not in the document]
Comment (hezhaozhao-git) [reply python/raydp/context.py:43]: yes

File: python/raydp/context.py [hunk 4]
```diff
@@ -148,6 +155,7 @@ def init_spark(app_name: str,
                num_executors: int,
                executor_cores: int,
                executor_memory: Union[str, int],
+               enable_hive: Optional[bool] = True,
```
[the remark this answers is not in the document]
Comment (hezhaozhao-git) [reply python/raydp/context.py:158]: ok

**Comment by hezhaozhao-git** [comment]
@kira-lin I didn't change the pytest file, why am I getting this error[error](https://github.com/oap-project/raydp/runs/5475262019?check_suite_focus=true)

**Comment by kira-lin** [comment]
@hezhaozhao-git Yes, it is not related to your pr. It's ok because linux CI passed.

**Comment by hezhaozhao-git** [comment]
> @kira-lin我没有更改 pytest 文件，为什么会出现此错误[错误](https://github.com/oap-project/raydp/runs/5475262019?check_suite_focus=true)

So I don't have to make any changes now, just wait for your next release right?

**Comment by kira-lin** [comment]
Can you please address the Optional issue mentioned above, and you can use our raydp-nightly release tomorrow. Thanks again

**Comment by hezhaozhao-git** [comment]
I don't know why this error occurs, so I don't know how to fix it，can you see more details? @kira-lin

**Comment by kira-lin** [comment]
Thanks, @hezhaozhao-git , I will fix the CI in other CIs, it's not related to your PR.

**Comment by kira-lin** [comment]
You can use raydp-nightly now, it includes this commit:)

**Pull request merged**